src/version/107/frontend/nl2go_manager.php

```diff
@@ -521,9 +521,9 @@ public function setSubscribe()
 			now()
 		);";
 
-		$GLOBALS["DB"]->executeQuery($query, 1);
+        $lastInsertId = $GLOBALS["DB"]->executeQuery($query, 7);
```

Comment:
and what is the meaning of 7 ? 😄  (maybe we should add comments, I'm pretty sure the next one will ask the same question again 😂 )

Reply:
After the query was executed JTLs NiceDB will try to set the return value based on this number. Here is the relevant snippet:

```
switch ($return) {
            case 1:
                $ret = $res->fetchObject();
                break;
            case 2:
                $ret = [];
                while (($row = $res->fetchObject()) !== false) {
                    $ret[] = $row;
                }
                break;
            case 3:
                $ret = $res->rowCount();
                break;
            case 7:
                $id = $this->pdo->lastInsertId();
                $ret = ($id > 0) ? $id : 1;
                break;
            case 8:
                $ret = $res->fetchAll(PDO::FETCH_NAMED);
                if (is_array($ret) && isset($ret[0])) {
                    $ret = $ret[0];
                } else {
                    $ret = null;
                }
                break;
            case 9:
                $ret = $res->fetchAll(PDO::FETCH_ASSOC);
                break;
            case 10:
                $ret = $res;
                break;
            case 11:
                $ret = $res->fetchAll(PDO::FETCH_BOTH);
                break;
            default:
                $ret = true;
                break;
        }
```

I will add descriptive constants for the returntypes for future use if thats ok with you.

Reply:
crazy shit oO thanks :)